osf/models/node.py

```diff
@@ -1484,15 +1485,21 @@ def set_privacy(self, permissions, auth=None, log=True, save=True, meeting_creat
                     return False
             self.is_public = True
             self.keenio_read_key = self.generate_keenio_read_key()
+            status = 'public'
         elif permissions == 'private' and self.is_public:
             if self.is_registration and not self.is_pending_embargo:
                 raise NodeStateError('Public registrations must be withdrawn, not made private.')
             else:
                 self.is_public = False
                 self.keenio_read_key = ''
+                status = 'unavailable'
         else:
             return False
 
+        for preprint in self.preprints.all():
+            if preprint.is_published:
+                update_ezid_metadata_on_change(preprint, status=status)
```

Comment:
This should be done in celery. We shouldn't allow an error from EZID prevent the project from becoming public/private.